h2/src/main/org/h2/table/Column.java

```diff
@@ -68,6 +68,7 @@
     private int displaySize;
     private Table table;
     private String name;
+    private int columnPos;
```

Comment:
1. Forget about Ignite - here we are making H2 better.
2. The approach looks wrong to me. As far I understand we must pass negative indexes for system columns.